pkg/processor/procbuiltin/unwrap.go

```diff
@@ -281,44 +291,60 @@ func (d *debeziumUnwrapper) validateRecord(data record.StructuredData) error {
 }
 
 func (d *debeziumUnwrapper) unwrapMetadata(rec record.Record) (record.Metadata, error) {
-	meta := record.Metadata{}
-	// add original record's metadata to meta
-	for k, v := range rec.Metadata {
-		meta[k] = v
-	}
-
-	structPayload := rec.Payload.After.(record.StructuredData)["payload"].(map[string]interface{})
-
-	// set metadata readAt time
-	var t time.Time
-	tsMs := structPayload[debeziumFieldTimestamp]
-	if tsMs != nil {
-		floatTime, ok := tsMs.(float64)
-		if !ok {
-			return nil, cerrors.Errorf("%s is not a float", debeziumFieldTimestamp)
+	debeziumRec := rec.Payload.After.(record.StructuredData)["payload"].(map[string]any)
+
+	var source map[string]string
+	for field, val := range debeziumRec {
+		switch field {
+		case debeziumFieldAfter, debeziumFieldBefore, debeziumFieldOp:
+			continue // ignore
+		case debeziumFieldTimestamp:
+			tsMs, ok := val.(float64)
+			if !ok {
+				return nil, cerrors.Errorf("%s is not a float", debeziumFieldTimestamp)
+			}
+			readAt := time.UnixMilli(int64(tsMs))
+			rec.Metadata.SetReadAt(readAt)
+		case debeziumFieldSource:
+			// don't add prefix for source fields
```

Comment:
What's the reason for this btw?

Reply:
For compatibility with the [SDK](https://github.com/ConduitIO/conduit-connector-sdk/blob/059da633a9feb696d0c61a27e806ed96252d1d29/record_formatter.go#L180). There we put metadata fields into `source`, so if another Conduit instance processes such a record it unwraps back the original OpenCDC record.

Reply:
Gotcha, thanks for the clarification! :raised_hands: Having a comment about that would be very useful IMO. :+1:

Reply:
Added the comment here: https://github.com/ConduitIO/conduit/pull/877/commits/30d21e0a651ede80eafe04cce9f3973d971ac6ce